dask_cuda/utils.py

```diff
@@ -649,3 +653,105 @@ def get_gpu_uuid_from_index(device_index=0):
     pynvml.nvmlInit()
     handle = pynvml.nvmlDeviceGetHandleByIndex(device_index)
     return pynvml.nvmlDeviceGetUUID(handle).decode("utf-8")
+
+
+def get_worker_config(dask_worker):
+    # assume homogenous cluster
+    plugin_vals = dask_worker.plugins.values()
+    ret = {}
+
+    # device and host memory configuration
+    for p in plugin_vals:
+        for v in dir(p):
+            # ignore hidden attrs as well as setup and CPUAffinity
+            if v.startswith("_") or v == "setup" or v == "cores":
+                pass
+            else:
+                ret[v] = getattr(p, v)
```

Comment:
This assumes that all plugins have disjoint attributes.

```
plugin_config = {}
for p in plugin_vals:
    plugin_config[type(p).__name__] = {v: getattr(p, v) for v in dir(p) if not (v.startswith("_") or v in {"setup", "cores"})}
```
(Untested)

Reply:
This is a lot clean and works nicely.  Thank you!

Reply:
Hmm, on second thought, I think it's better to key something like I had before but flatten the key name like:

> key = type(p).__name__ + '.' + v

This makes it easier to read without having to scan a nested dictionary within the table

Reply:
> This makes it easier to read without having to scan a nested dictionary within the table

Or one could translate (in the rich output), like so:

```python
from functools import singledispatch
from rich.pretty import Pretty
from rich.table import Table
from operator import itemgetter

@singledispatch
def pretty(obj):
    return Pretty(obj)

@pretty.register(dict)
def pretty_dict(obj):
    t = Table()
    t.add_column("key")
    t.add_column("value")
    for k, v in sorted(obj.items(), key=itemgetter(0)):
        t.add_row(pretty(k), pretty(v))
    return t

...

data = get_config(...)
Console().print(pretty(data))
```

WDYT?

Reply:
👍  - this is really cool!